### PR TITLE
feat(Flightradar24): add presence

### LINKED ISSUE #5486: Flightradar24 | flightradar24.com
### Discussed in https://github.com/PreMiD/Presences/discussions/5485

<div type='discussions-op-text'>

<sup>Originally posted by **developedbyalex** January  3, 2022</sup>
Prerequisites and essential questions

- [x] Is it a popular site?
- [x] Is the website older than 2 months?
- [ ] Is the site locked to a specific country/region?
- [ ] Is the site a paid service? (e.g. Netflix, Hulu)
- [ ] Does the website feature NSFW content? (e.g. porn, etc...)
- [ ] Are you a donator/patron?
- [x] Do you acknowledge that coding presences is completely voluntary and may take time for your service to be added regardless of priority?

**What's your Discord username?**
ᴀʟᴇx#0001

**What's the name of the service?**
https://www.flightradar24.com/ I Flightradar24
**What should the Presence display?**
If they're tracking a flight, if so what flight they're tracking, the airport they're going from and landing. If not, just "viewing map"

If possible, please provide a logo for the service (512x512 minimum)
https://i.imgur.com/k0iE2zY.png</div>

### PR DESCRIPTION
**Pull request by joerkig**
**Describe the changes in this pull request:**

Add Flightradar24 presence

Please do point me in the right direction if I did something in completely the worst/wrong way ❤️ 

Resolves #5486 

<details>
<summary> Proof showing the creation/modification is working as expected </summary>
<br>

Discord_w27I6u1Wse
Discord_Oni2rVFPYz
Discord_ij64FYmWxW
Discord_EEVbQPE4ZI
Discord_mfX7SSdgoC
Discord_xq7c0Q2VVm
Discord_0XcPCbIj5v
Discord_Ui2QnbItwy
Discord_777OG7MyXP
Discord_hr5cFxvxpo
Discord_LhCSLUTETl
Discord_iR61L3gj2V
Discord_b7tBChPRcD
Discord_nfpabhjQZV
Discord_FHzqLVhne8
Discord_WckZsuwiTY
Discord_rOvaXvsKgy
Discord_t5AdL7QYs2
Discord_0lPacg2ToA
Discord_Ol6DLG7Fe5
Discord_mI9AtksbNk
Discord_OsUc8tSe5L
Discord_gLkFDNlaLy
Discord_E45wEW4gFC
Discord_uBir3a0SO9
Discord_wJmYzmBXjL
Discord_nawFXflvDw
Discord_lqp4NHvX4P
Discord_4xRtyyevts
Discord_ChjqGh6g1Q
Discord_KED8gH2Gev
Discord_mQSZS34tjn
Discord_2Y7fq16not
Discord_k3BCaufuGf
Discord_h32sNbVfKT
Discord_A7Uosdljvk
Discord_2wXJD29nyX
Discord_CF0zLx08sp
Discord_9vjIpEVVmu
Discord_8cALomYEAk

</details>

### CODE REVIEW

File: websites/F/Flightradar24/iframe.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+let total: string,
+	elapsed: string,
+	isPlaying: boolean,
+	currentTime: number,
+	duration: number,
+	paused: boolean;
+
+if (document.location.href.includes("player.blubrry.com")) {
+	iframe.on("UpdateData", async () => {
+		//Get all the data you need out of the iFrame save them in variables and then send them using iframe.send
```
Comment (EncryptedDev) [inline websites/F/Flightradar24/iframe.ts:11]: Remove comments across this entire file

File: websites/F/Flightradar24/presence.ts [hunk 2]
```diff
@@ -0,0 +1,690 @@
...
+			});
+	}
+);
+
+presence.on("UpdateData", async () => {
+	const [elapsed, buttons, images, left] = await Promise.all([
+			presence.getSetting<boolean>("elapsed"),
+			presence.getSetting<boolean>("buttons"),
+			presence.getSetting<boolean>("images"),
+			presence.getSetting<boolean>("left")
```
Comment (EncryptedDev) [inline websites/F/Flightradar24/presence.ts:46]: `left`, what does that mean?

File: websites/F/Flightradar24/presence.ts [hunk 3]
```diff
@@ -0,0 +1,690 @@
...
+									.toUpperCase()}${document
+									.querySelector("small")
+									.textContent.split(" ")[2]
+									.slice(1)}`;
+							} else if (document.querySelector(".airport-name")) {
+								presenceData.details = `Viewing ${
+									document.querySelector(".airport-name").textContent
+								}`;
+								presenceData.state =
+									document.querySelector<HTMLElement>(".btn.active").innerText;
```
Comment (EncryptedDev) [inline websites/F/Flightradar24/presence.ts:80]: Don't use HTMLElement, find the proper element type

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

File: websites/F/Flightradar24/presence.ts (hunk 2 above)
Comment (joerkig) [reply websites/F/Flightradar24/presence.ts:46]: > `left`, what does that mean?

NVIDIA_Share_ox4VHXLccj

File: websites/F/Flightradar24/presence.ts [hunk 4]
```diff
@@ -0,0 +1,695 @@
...
+						presenceData.details = "Viewing Newsletter";
+					else if (document.location.pathname === "/blog/avtalk-podcast/")
+						presenceData.details = "Viewing Podcast";
+					else if (document.querySelector("iframe.blubrryplayer")) {
+						presenceData.details = document
+							.querySelector(".elementor-heading-title.elementor-size-default")
+							.textContent.split(":")[0];
+						presenceData.state = document
+							.querySelector(".elementor-heading-title.elementor-size-default")
+							.textContent.split(":")[1];
```
Comment (Slowlife01) [inline websites/F/Flightradar24/presence.ts:286]:
```suggestion
						[presenceData.details, presenceData.state] = document
							.querySelector(".elementor-heading-title.elementor-size-default")
							.textContent.split(":");
```

**Review by EncryptedDev — APPROVED** [review]
lgtm

**Comment by Bas950** [comment]
Would be cool if you add an option to use the selected planes image as cover/largeImageKey

**Comment by joerkig** [comment]
> Would be cool if you add an option to use the selected planes image as cover/largeImageKey

Forgot to take screenshots with the option on

Discord_p4waRathzD
Discord_5BY8XctN4X

**Review by Slowlife01 — APPROVED** [review]
lgtm

**Pull request merged**